channel_print.py

```diff
@@ -0,0 +1,314 @@
+        gid = str(msg.guild.id)
+
+        if msg.attachments:
+            _file = BytesIO()
+
+            await msg.attachments[0].save(_file)
+            try:
+                wall = decode_json(_file.getvalue())
+            except ValueError as e:
+                self.logger.exception("Could not decode uploaded document file!", exc_info=True)
+                raise CommandSyntaxError(e)
+            except Exception as e:
+                raise CommandSyntaxError(f"Not a valid JSON file: {e}")
+
+            try:
+                wall = verify_document(wall)
+            except ValueError as e:
+                raise CommandSyntaxError(e)
+        else:
+            if gid not in self.walls:
+                self.walls[gid] = dict()
+                return
+
+            try:
+                wall = msg.clean_content.split(None, 1)[1]
+            except IndexError:
+                raise CommandSyntaxError
+
+            if wall not in self.walls[gid]:
+                raise CommandSyntaxError("No such document.")
+
+            try:
+                wall = verify_document(self.walls[gid][wall])
+            except ValueError as e:
+                await respond(msg, f"**WARNING: {e}**")
+                return
+
+        for post in wall:
+            if isinstance(post, str):
+                await respond(msg, post)
+            else:
+                _file = None
+                if 'file' in post:
+                    try:
+                        _file = urlopen(post['file'])
+                        if int(_file.info()['Content-Length']) > self.plugin_config['max_filesize']:
+                            raise ValueError("File too big.")
+                        _file = File(_file,
+                                     filename="wallfile" + mimetypes.guess_extension(_file.info()['Content-Type']))
+                    except Exception as e:
+                        self.logger.info(f"Attachment file error in {msg.guild}:\n{e}")
+                        await self.plugin_manager.hook_event("on_log_event", msg.guild,
+                                                             f"**WARNING: Error occured during printout:**\n{e}",
+                                                             log_type="print_event")
+                        _file = None  # we just want to fail quietly, files are finnicky
+                if _file or 'embed' in post or 'content' in post:
+                    await respond(msg,
+                                  post.get('content', None),
+                                  embed=EmbedDict(post['embed']) if 'embed' in post else None,
+                                  file=_file)
+
+    @Command("DeletePrint", "PrintDelete",
+             doc="Deletes the specified document.",
+             syntax="(document)",
+             perms={"manage_messages"},)
+    async def _deleteprint(self, msg):
+        gid = str(msg.guild.id)
+
+        try:
+            name = msg.clean_content.split(None, 2)[1].lower()
```

Comment:
Should probably be a `.split(None, 1)`

Reply:
This is done to extract a single word from something like `"!cmd WORD all the other words we don't want here"`. Document IDs are not allowed to have spaces in them, and this ensures that.